FastQC
------

Racon is intended as a standalone consensus module to correct raw contigs generated by rapid assembly methods which do not include a consensus step. The goal of Racon is to generate genomic consensus which is of similar or better quality compared to the output generated by assembly methods which employ both error correction and consensus steps, while providing a speedup of several times compared to those methods. It supports data produced by both Pacific Biosciences and Oxford Nanopore Technologies.

Racon can be used as a polishing tool after the assembly with either Illumina data or data produced by third generation of sequencing. The type of data inputed is automatically detected.

Racon takes as input only three files: contigs in FASTA/FASTQ format, reads in FASTA/FASTQ format and overlaps/alignments between the reads and the contigs in MHAP/PAF/SAM format. Output is a set of polished contigs in FASTA format printed to stdout. All input files can be compressed with gzip (which will have impact on parsing time).


We are going to use racon to do an initial correction. It is advised to do one or two rounds with racon before polishing with medaka.

  cd ~/workdir/
  mkdir nanopore_mapping
  cd ~/workdir/nanopore_mapping/
  samtools view -S mapping.sorted.bam > mapping.sorted.sam
  cd ~/workdir/
  mkdir racon
  racon -t 14 ~/workdir/basecall/basecall.fastq.gz ~/workdir/nanopore_mapping/mapping.sorted.sam ~/workdir/assembly/assembly.contigs.fasta > racon/racon.fasta

  
References
^^^^^^^^^^

**racon** https://github.com/isovic/racon
